catroid/src/main/java/org/catrobat/catroid/sensing/CollisionInformation.java

```diff
@@ -79,73 +83,102 @@ public int getNumberOfVertices() {
 		return size;
 	}
 
+	private void calculateBubblePositions(Bitmap source) {
+		int[] pixels = new int[source.getWidth() * source.getHeight()];
+		source.getPixels(pixels, 0, source.getWidth(), 0, 0, source.getWidth(), source.getHeight());
+
+		Integer centerX = source.getWidth() / 2;
+		Integer centerY = source.getHeight() / 2;
+
+		for (int y = 0; y < source.getHeight(); y++) {
+			if (rightBubblePos != null) {
+				return;
+			}
+			for (int x = source.getWidth() - 1; x > 0; x--) {
+				if (!ImageEditing.isPixelTransparent(source, pixels, x, y)) {
+					Integer xDiff = x - centerX;
+					Integer yDiff = centerY - y;
+
+					if (rightBubblePos == null) {
+						rightBubblePos = new Pair<>(xDiff, yDiff);
+					}
+
+					leftBubblePos = new Pair<>(xDiff, yDiff);
+				}
+			}
+		}
+	}
+
 	public void loadOrCreateCollisionPolygon() {
```

Comment:
I would prefer this function to be split up into smaller functions

Reply:
Naming also implies this - There should never be an OR or an AND in a function name.

Reply:
I didn't write this function, just modified it a little. But I can try to split it up.

Reply:
Changing this function would result in modifyng four files and four tests.
Should I do it nonetheless?